src/Proto.Actor/Future/SharedFuture.cs

```diff
@@ -70,7 +70,15 @@ internal SharedFutureProcess(ActorSystem system, int size) : base(system)
         private PID Pid { get; }
         public bool Stopping { get; private set; }
 
-        public int RequestsInFlight => (int) (_createdRequests - _completedRequests);
+        public int RequestsInFlight {
+            get {
+                // Read completedRequests first and createdRequests later so that we will
+                // never read the 2 vars in an order that would result in completedRequests > createdRequests.
+                long completed = Interlocked.Read(ref _createdRequests);
```

Comment:
looks like here should be _completedRequests.

Reply:
Yes, you are correct. thanks for spotting it

Reply:
Seems like this is already handled by mhelleborg. Resolving this.